core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -441,6 +442,21 @@ private void checkDepthLimit(String objectPath, long maxDepth) {
         }
     }
 
+    /**
+     * We only allow upto 1 percolator field per index.
+     *
+     * Reasoning here is that the PercolatorQueryCache only supports a single document having a percolator query.
+     * Also specifying multiple queries per document feels like an anti pattern
+     */
+    private void checkPercolatorFieldLimit(List<FieldMapper> fieldMappers) {
+        long numPercolatorFields = fieldMappers.stream()
+                .filter(fieldMapper -> fieldMapper instanceof PercolatorFieldMapper)
+                .count();
+        if (numPercolatorFields > 1) {
+            throw new IllegalArgumentException("Up to one percolator field type is allowed per index");
+        }
```

Comment:
it will allow to use a percolator field across several types, but I think it is fine, and it plays well with the fact that we are progressively divorcing types from mappings